app/components/sidebars/main/channels_list/list/index.js

```diff
@@ -2,46 +2,69 @@
 // See LICENSE.txt for license information.
 
 import {connect} from 'react-redux';
+import {createSelector} from 'reselect';
 
 import {General} from 'mattermost-redux/constants';
 import {
     getSortedUnreadChannelIds,
-    getSortedFavoriteChannelIds,
-    getSortedPublicChannelIds,
-    getSortedPrivateChannelIds,
-    getSortedDirectChannelIds,
+    getOrderedChannelIds,
 } from 'mattermost-redux/selectors/entities/channels';
 import {getCurrentUserId, getCurrentUserRoles} from 'mattermost-redux/selectors/entities/users';
 import {getCurrentTeamId} from 'mattermost-redux/selectors/entities/teams';
 import {getTheme, getFavoritesPreferences} from 'mattermost-redux/selectors/entities/preferences';
+import {getSidebarPreferences} from 'mattermost-redux/selectors/entities/sidebar';
 import {showCreateOption} from 'mattermost-redux/utils/channel_utils';
 import {isAdmin as checkIsAdmin, isSystemAdmin as checkIsSystemAdmin} from 'mattermost-redux/utils/user_utils';
 import {getConfig, getLicense} from 'mattermost-redux/selectors/entities/general';
 
+import {SidebarSectionTypes} from 'app/constants/view';
+
 import List from './list';
 
+const filterZeroUnreads = createSelector(
+    (sections) => sections,
+    (sections) => {
+        return sections.filter((s) => {
+            if (s.type === SidebarSectionTypes.UNREADS) {
+                return s.items.length > 0;
+            }
+            return true;
+        });
+    }
+);
+
 function mapStateToProps(state) {
     const config = getConfig(state);
     const license = getLicense(state);
     const roles = getCurrentUserId(state) ? getCurrentUserRoles(state) : '';
-    const unreadChannelIds = getSortedUnreadChannelIds(state);
-    const favoriteChannelIds = getSortedFavoriteChannelIds(state);
-    const privateChannelIds = getSortedPrivateChannelIds(state);
-    const directChannelIds = getSortedDirectChannelIds(state);
     const currentTeamId = getCurrentTeamId(state);
-    const publicChannelIds = getSortedPublicChannelIds(state);
-
     const isAdmin = checkIsAdmin(roles);
     const isSystemAdmin = checkIsSystemAdmin(roles);
+    const sidebarPrefs = getSidebarPreferences(state);
+    const lastUnreadChannel = state.views.channel.keepChannelIdAsUnread;
+    const unreadChannelIds = getSortedUnreadChannelIds(state, lastUnreadChannel);
```

Comment:
In the webapp makes sense cause the sidebar is always visible and we want to prevent the jump around of the channel where the user can get lost, but why do we want to keep the last unread channel here? the sidebar on mobile will always close so the jump is never visible.

Reply:
You're right, looks like we don't need it. Will remove.